database/postgresql/src/main/java/com/example/javaopencommerce/warehouse/PsqlWarehouseItemRepositoryImpl.java

```diff
@@ -0,0 +1,58 @@
+package com.example.javaopencommerce.warehouse;
+
+import com.example.javaopencommerce.exception.EntityNotFoundException;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+import java.util.List;
+
+@ApplicationScoped
+class PsqlWarehouseItemRepositoryImpl implements PsqlWarehouseItemRepository {
+
+    private final EntityManager em;
+
+    public PsqlWarehouseItemRepositoryImpl(EntityManager em) {
+        this.em = em;
+    }
+
+    @Override
+    public ItemStockEntity getItemById(Long id) {
+        try {
+            return em.createQuery("SELECT i FROM ItemStock i WHERE i.id = ?1", ItemStockEntity.class)
+                    .setParameter(1, id)
+                    .getSingleResult();
+        } catch (NoResultException e) {
+            throw new EntityNotFoundException("Order", id);
+        }
+    }
+
+    @Override
+    public List<ItemStockEntity> getItemStocksByIdList(List<Long> ids) {
+        return em.createQuery("SELECT i FROM ItemStock i WHERE i.id in(?1)", ItemStockEntity.class)
+                .setParameter(1, ids)
+                .getResultList();
+    }
+
+    @Override
+    public void updateStock(Long id, Integer newStock) {
+        int result = em.createQuery("UPDATE ItemStock i SET i.stock = ?1 WHERE i.id = ?2")
+                .setParameter(1, newStock)
+                .setParameter(2, id)
+                .executeUpdate();
+
+        if (result != 1) {
+            throw new IllegalStateException("Failed to update stocks for item with id: " + id);
```

Comment:
It would be good to throw a dedicated exception here as well.
```suggestion
// TODO: thorw a dedicated exception
            throw new IllegalStateException("Failed to update stocks for item with id: " + id);
```